sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -4,44 +4,77 @@
 package akka.stream.alpakka.sqs
 
 import java.util
+import java.util.concurrent.Executors
 
 import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
 import akka.stream.{Attributes, Outlet, SourceShape}
+import com.amazonaws.auth.{AWSCredentials, DefaultAWSCredentialsProviderChain}
 import com.amazonaws.handlers.AsyncHandler
 import com.amazonaws.services.sqs.AmazonSQSAsyncClient
 import com.amazonaws.services.sqs.model.{Message, ReceiveMessageRequest, ReceiveMessageResult}
 
 import scala.collection.JavaConverters._
 
 object SqsSourceSettings {
-  val Defaults = SqsSourceSettings(20, 100, 10)
+  val Defaults = SqsSourceSettings(20, 100, 10, None)
+
+  def create(waitTimeSeconds: Int,
+             maxBufferSize: Int,
+             maxBatchSize: Int,
+             credentials: AWSCredentials): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize, Some(credentials))
+
+  private def create(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int): SqsSourceSettings =
+    SqsSourceSettings(waitTimeSeconds, maxBufferSize, maxBatchSize, None)
+
 }
 
 //#SqsSourceSettings
-final case class SqsSourceSettings(waitTimeSeconds: Int, maxBufferSize: Int, maxBatchSize: Int) {
+final case class SqsSourceSettings(
+    waitTimeSeconds: Int,
+    maxBufferSize: Int,
+    maxBatchSize: Int,
+    credentials: Option[AWSCredentials]
+) {
   require(maxBatchSize <= maxBufferSize)
   // SQS requirements
   require(waitTimeSeconds >= 0 && waitTimeSeconds <= 20)
   require(maxBatchSize >= 1 && maxBatchSize <= 10)
 }
 //#SqsSourceSettings
 
-final class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings, sqsClient: AmazonSQSAsyncClient)
-    extends GraphStage[SourceShape[Message]] {
+final class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings) extends GraphStage[SourceShape[Message]] {
 
   val out: Outlet[Message] = Outlet("SqsSource.out")
   override val shape: SourceShape[Message] = SourceShape(out)
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
     new GraphStageLogic(shape) {
 
+      private val credentials =
+        settings.credentials.getOrElse(DefaultAWSCredentialsProviderChain.getInstance().getCredentials)
+
+      private val maxConcurrency = settings.maxBufferSize / settings.maxBatchSize
+      private val threadPool = Executors.newFixedThreadPool(maxConcurrency)
```

Comment:
The javadoc for `AmazonSQSAsyncClient` says following: 

> Shuts down the client, releasing all managed resources.This includes forcibly terminating all pending asynchronous service calls.

Should I rely on that or shutdown everything manually?